presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -134,6 +137,12 @@ private RubixConfigurationInitializer initializeRubix(RubixConfig rubixConfig)
 
         // initialize rubix in master-only mode
         rubixConfig.setStartServerOnCoordinator(true);
+
+        // use different rubix ports for each test to prevent nondeterministic bind errors
+        rubixConfig.setBookKeeperServerPort(DEFAULT_BOOKKEEPER_SERVER_PORT + cachingFileSystemCounter);
```

Comment:
I would rather add some logic to probe for port to see if it can be bound. Not just blindly iterate forward. We can run into some other port which is already open on the machine.
Also I want to understand what is the reason for the ports to remain open after previous tests complete. Maybe we can fix the root cause.

Reply:
added waiting for Rubix servers to stop